radix-engine/src/model/vault.rs

```diff
@@ -56,39 +49,44 @@ impl Vault {
         &mut self,
         ids: &BTreeSet<NonFungibleId>,
     ) -> Result<Bucket, VaultError> {
-        let this_container = self.borrow_container()?;
-
         Ok(Bucket::new(
-            this_container
+            self.borrow_container_mut()
                 .take_non_fungibles(ids)
                 .map_err(VaultError::ResourceContainerError)?,
         ))
     }
 
-    pub fn liquid_amount(&self) -> ResourceAmount {
-        self.container.liquid_amount()
-    }
-
     pub fn resource_def_id(&self) -> ResourceDefId {
-        self.container.resource_def_id()
+        self.borrow_container().resource_def_id()
     }
 
     pub fn resource_type(&self) -> ResourceType {
-        self.container.resource_type()
+        self.borrow_container().resource_type()
     }
 
-    /// Creates another `Rc<ResourceContainer>` to the container
-    pub fn reference_container(&self) -> Rc<ResourceContainer> {
-        self.container.clone()
+    pub fn total_amount(&self) -> Decimal {
+        self.borrow_container().total_amount()
+    }
+
+    pub fn total_ids(&self) -> Result<BTreeSet<NonFungibleId>, VaultError> {
+        self.borrow_container()
+            .total_ids()
+            .map_err(VaultError::ResourceContainerError)
+    }
+
+    pub fn is_locked(&self) -> bool {
+        self.borrow_container().is_locked()
     }
 
-    /// Creates a mutable reference to the container
-    pub fn borrow_container(&mut self) -> Result<&mut ResourceContainer, VaultError> {
-        Ok(Rc::get_mut(&mut self.container).ok_or(VaultError::VaultLocked)?)
+    pub fn borrow_container(&self) -> Ref<ResourceContainer> {
+        self.container.borrow()
     }
 
-    /// Takes the ownership of the container
-    pub fn take_container(self) -> Result<ResourceContainer, VaultError> {
-        Ok(Rc::try_unwrap(self.container).map_err(|_| VaultError::VaultLocked)?)
+    pub fn borrow_container_mut(&mut self) -> RefMut<ResourceContainer> {
```

Comment:
Shouldn't be `pub`

Reply:
I was debating whether I should expose these "proxy" getter methods or a reference to the row resource container. 

As commented in another thread, if we decide to remove these getters, we will have to expose the borrow methods; if not, I'm totally happy to make them private.